walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -22,20 +22,40 @@
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    List<TimeRange> unavailableTimes = this.getUnavailableTimes(events, request);
-    return this.getAvailableTimes(events, request, unavailableTimes);
+    Collection<String> mandatoryAttendees = request.getAttendees();
+    List<TimeRange> unavailableTimes;
+    Collection<TimeRange> availableTimes;
+    if (mandatoryAttendees.size() > 0){
+      //Get time ranges for which mandatory attendees are unavailable.
+      unavailableTimes = this.getUnavailableTimes(events, mandatoryAttendees);
+      //Get time ranges for which mandatory attendees are available.
+      availableTimes = this.getAvailableTimes(events, request, unavailableTimes);
```

Comment:
You don't want to compute this unless at the very end `availableTimesWithOptional.size() > 0` is false right? I would move this computation to the end to avoid calculating this when it's not used.

Reply:
Oh yes! this is a very good point